### ISSUE TITLE
Aria-label “edit this reflection” should not exist when the reflection is no longer editable

### ISSUE THREAD

**Issue by acressall** [ParabolInc/parabol#7107]
## Issue - Bug
Reflection cards maintain the editing markup even when the reflections are no longer editable. The biggest issue here is we have an area label that says "edit this reflection" when it is not editable.
This problem for group, vote, and discussion phases.

### Acceptance Criteria (optional)
Users can:
   - no longer be told that a div allows them to edit a reflection when editing is disabled
Triage is performed to find the root cause of the bug, timeboxed to ~1 hour. 

**Estimated effort:** 1 hour to triage. More if root cause is already identified.

**Comment by 87prashant** [comment]
@acressall 
Is this creating any issues?
I was checking with a screen reader and I was not able to find any issue as it reads `edit this reflection` only when the text is editable.
Or am I missing something?

**Comment by enriquesanchez** [comment]
I'm using VoiceOver on the Mac, and the text `Edit this reflection` is announced to screen readers when the user moves their VoiceOver focus to the reflection card, regardless of what phase the meeting is in. 

In the `Group` phase users can edit their own reflections (but not the ones created by others), and on the`Vote` and `Discuss` phases reflections are no longer editable, regardless of who the author is.

**Comment by 87prashant** [comment]
Ok, thanks. BTW I am testing with NVDA
